File: Algorithms/Implementation/Encryption.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <math.h>
using namespace std;


int main(){
    string s;
    cin >> s;
    float len = s.length();
    float col = ceil(sqrt(len));
    int row = ceil(len/col);
    for(int i=0; i<col; i++)
    {
        for(int j=0; j<row; j++)
        {
            if(i+(j*col)<len)
            {
                cout<<s[i+(j*col)];
            }
        }
        cout<<" ";
    }
    return 0;
}

File: Algorithms/Implementation/Encryption.py
#!/bin/python3

import sys,math


s = input().strip()

col = math.ceil(pow(len(s),.5))

for i in range(col):
    for j in range(math.ceil(len(s)/col)):
        try:
            print(s[i+(j*col)],end="")
        except:
            print("",end="")
    print(" ",end="")
